app/src/components/notification/Notification.tsx

```diff
@@ -1,48 +1,198 @@
-import React from "react";
-import { useDispatch } from "react-redux";
+import React, { useEffect, useRef, useState } from "react";
+import { shallowEqual, useDispatch, useSelector } from "react-redux";
 import { Link } from "react-router-dom";
-import { markNotificationAsReadAsync } from "../../pages/notifications/notificationSlice";
+import {
+  dismissNotificationAsync,
+  markNotificationAsReadAsync,
+  selectOpenMenus,
+  setCloseContextMenu,
+  setContextMenuOpen,
+} from "../../pages/notifications/notificationSlice";
 
 import Avatar from "../avatar";
 import ContextMenuButton from "./context-menu-button";
 import "./notification-component-style.css";
 import { INotificationCardData } from "./notification.types";
+import { useSwipeable } from "react-swipeable";
+import NotificationContextMenu from "./context-menu";
+import { NotificationContextMenuAction } from "./context-menu/Notification-context-menu";
+import NotificationActionConfirmation from "./notification-confirmation-action-modal";
 
 function NotificationElement(props: INotificationCardData) {
   const dispatch = useDispatch();
+  const [isSwiping, setIsSwiping] = useState<boolean>(false);
+  const [swipingObject, setSwipingObject] = useState<any>();
+  const [contextMenuVisible, setContextMenuVisible] = useState<boolean>(false);
+  const [showConfirmation, setShowConfirmation] = useState<boolean>(false);
+  const [
+    confirmationAction,
+    setConfirmationAction,
+  ] = useState<NotificationContextMenuAction | null>(null);
+  const [confirmationMessage, setConfirmationMessage] = useState<string>("");
 
   const handleOnLinkClick = (notificationId: string) => {
-    // Here we need to mark the notification as read
     dispatch(markNotificationAsReadAsync(notificationId));
   };
+  const openMenus = useSelector(selectOpenMenus, shallowEqual);
+
+  useEffect(() => {
+    if (swipingObject) {
+      swipingObject.style["margin-left"] = `-1300px`;
+      if (props.id && isSwiping) {
+        dispatch(dismissNotificationAsync(props.id));
+        setIsSwiping(false);
+      }
+    }
+  }, [dispatch, swipingObject, props.id, isSwiping]);
+
+  const swipeHandlers = useSwipeable({
+    preventDefaultTouchmoveEvent: true,
+    trackMouse: true,
+    trackTouch: true,
+    onSwipedLeft: (data: any) => {
+      if (!isSwiping && data.velocity < 1) {
+        if (data.event.currentTarget) {
```

Comment:
```suggestion
        if (data.event?.currentTarget?.style) {
```

Reply:
Without this check I still get `TypeError: Cannot set property 'margin-left' of undefined`
To reproduce, with Chrome's device toolbar switched off, click and drag text to left